src/boot/bootctl-install.c

```diff
@@ -224,15 +251,25 @@ static int copy_file_with_version_check(const char *from, const char *to, bool f
         if (fd_from < 0)
                 return log_error_errno(errno, "Failed to open \"%s\" for reading: %m", from);
 
-        if (!force) {
+        if (!FLAGS_SET(flags, COPY_WITH_VERSION_CHECK_FORCE)) {
                 fd_to = open(to, O_RDONLY|O_CLOEXEC|O_NOCTTY);
                 if (fd_to < 0) {
                         if (errno != ENOENT)
                                 return log_error_errno(errno, "Failed to open \"%s\" for reading: %m", to);
                 } else {
-                        r = version_check(fd_from, from, fd_to, to);
-                        if (r < 0)
-                                return r;
+                        if (FLAGS_SET(flags, COPY_WITH_VERSION_CHECK_HASH_ONLY)) {
```

Comment:
so i think we should always compare the files and avoid a copy if the files already match.

hence, please just call a compare_data_fd() or so here unconditionally. And if the files match then just shortcut the copy always.

And then provide a flag COPY_WITH_VERSION_CHECK_IGNORE_VERSION or so, that simply disables the version_check() call